### ISSUE TITLE
User view empty if using ldap3 2.1.1 and objectGUID as primary login attribute

### ISSUE THREAD

**Issue by fredreichbier** [privacyidea/privacyidea#1083]
Steps to reproduce:
* Use current master (6f082de)
* Install ldap3 2.1.1, pyasn1 0.1.9
* Set up LDAP resolver with login name attribute ``objectGUID``
* Go to an arbitrary user detail view. It is empty, even though the token list is displayed correctly.

This is because ``getUserList`` does not encode the objectGUID UUID string to a byte string before adding it to the search filter:
https://github.com/privacyidea/privacyidea/blob/6f082deeef8775e50f8be447b431493e209edfb8/privacyidea/lib/resolvers/LDAPIdResolver.py#L610

In other words, it is missing a ``trim_objectGUID``.

This works with ldap3 2.5, because this version also supports UUID strings in search filters.

**Comment by cornelinux** [comment]
In most cases we do not use 2.1.1 anymore:
The enterprise repo is updated to 2.5
The launchpad 16.04 repos are updated 2.5
Only the trusty launchpad repo 14.04 could still use 2.1.1 (Those should upgrade to 16.04)

Should we simply add a ``trim_objectGUID`` or should we simply close the ticket?

**Comment by fredreichbier** [comment]
Closing this because
* we hopefully won't encounter ldap3 2.1.1 in the wild anymore
* objectGUID as login attribute didn't work in privacyIDEA 2.22 anyway (see #1076), so there is no point in supporting legacy installations with ldap3 2.1.1
